db/compaction/compaction.cc

```diff
@@ -1020,7 +1022,7 @@ int Compaction::EvaluatePenultimateLevel(
   }
 
   bool supports_per_key_placement =
-      immutable_options.preclude_last_level_data_seconds > 0;
+      mutable_cf_options.preclude_last_level_data_seconds > 0;
```

Comment:
`Compaction` makes a copy of the mutable column family options in `mutable_cf_options_`.  Could that option be used here?